server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v70/PopulateDefaultQualityGateTest.java

```
@@ -0,0 +1,143 @@
+  }
+
+  @Test
+  public void should_populate_defaultQualityGate_column() throws SQLException {
+    String builtInQGUuid = Uuids.createFast();
+    insertQualityGate(builtInQGUuid, "Sonar way", true);
+    String orgUuid1 = Uuids.createFast();
+    String orgUuid2 = Uuids.createFast();
+    insertOrganization(orgUuid1);
+    insertOrganization(orgUuid2);
+
+    underTest.execute();
+
+    // all organizations have the builtIn quality gate
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid != '" + builtInQGUuid + "'")
+    ).isEqualTo(0);
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid = '" + builtInQGUuid + "'")
+    ).isEqualTo(2);
+
+    // updated_at must have been updated
+    assertThat(
+      db.countSql("select count(uuid) from organizations where updated_at = " + NOW )
+    ).isEqualTo(2);
+  }
+
+  @Test
+  public void is_reentrant() throws SQLException {
+    String builtInQGUuid = Uuids.createFast();
+    insertQualityGate(builtInQGUuid, "Sonar way", true);
+    insertOrganization(Uuids.createFast());
+    insertOrganization(Uuids.createFast());
+
+    underTest.execute();
+    underTest.execute();
+
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid != '" + builtInQGUuid + "'")
+    ).isEqualTo(0);
+
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid = '" + builtInQGUuid + "'")
+    ).isEqualTo(2);
+  }
+
+  @Test
+  public void should_fail_with_ISE_when_no_builtIn() throws SQLException {
+    insertOrganization(Uuids.createFast());
+
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("Unable to find the builtin quality gate");
+
+    underTest.execute();
+  }
+
+  @Test
+  public void should_not_fail_if_there_is_multiple_builtin_qualitygates() throws SQLException {
```

If it would have been me, I would fail fast in this case => This is a strange situation where it's hard to predict what will happen